src/pages/cart.jsx

```diff
@@ -23,13 +23,97 @@ import {
 import { Seo } from "../components/seo"
 
 export default function CartPage() {
-  const { checkout, loading } = React.useContext(StoreContext)
+  const { checkout, loading, client, setCheckout } = React.useContext(StoreContext)
   const emptyCart = checkout.lineItems.length === 0
 
-  const handleCheckout = () => {
+  const handleCheckout = async () => {
+    if(window.onwardApp) {
+      await window.onwardApp.updateCartForCheckout();
+    }
     window.open(checkout.webUrl)
   }
 
+  const parseCheckout = (checkout) => {
+    return {
+      currency: checkout.currencyCode || null,
+      items: checkout.lineItems.map(item =>
+        ({
+          id: item.id,
+          cost: { amount: Number(item.variant.price.amount) * item.quantity, currencyCode: item.variant.price.currencyCode },
+          quantity: item.quantity,
+          properties: null,
+          variant: {
+            price: null,
+            requiresShipping: true,
+            id: item.variant.id,
+          }
+        })
+      ) || []
+    };
+  };
+
+  React.useEffect(() => {
+    if(window.onwardApp) {
+      window.onwardApp.cartChanged(parseCheckout(checkout));
+    }
+  }, [checkout]);
```

Comment:
If the checkout object changes (increment/decrement quantities) this triggers and  updates the onward cart